frontend/src/pages/Wallet.js

```diff
@@ -0,0 +1,143 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2021 
+
+import {
+	Button,
+	Table,
+	TableBody,
+	TableCell,
+	TableContainer,
+	TableRow,
+} from "@mui/material";
+import React, { useEffect } from "react";
+import appwriteApi from "../api/appwriteApi";
+import { useHistory } from "react-router-dom";
+import Grid from "@mui/material/Grid";
+import { useState } from "react";
+import detectEthereumProvider from "@metamask/detect-provider";
+import CenterFlexBoxMedium from "../components/CenterFlexBoxMedium";
+
+function AccountEntry({ data }) {
+	return 	<TableRow>
+		<TableCell style={{ color: "white", borderBottom: "none" }}>{data.pubAddres}</TableCell>
+		<TableCell style={{ color: "white", borderBottom: "none" }}>{parseInt(data.bl, 16)}</TableCell>
+	</TableRow>;
+}
+
+function EthAccountsDetail({ publicAddresses }) {
+	const [detailData, setDetailData] = useState([]);
+
+	useEffect(() => {
+		async function fetchAccountsDetails(publicAddresses) {
+			const ethereum = await detectEthereumProvider();
+			if (!ethereum) {
+				console.error("no ethereum");
+				return;
+			}
+			let data = [];
+			// This is the most bullshit thing I've seen. Can't use ForEach nor For...in!
+			const length = publicAddresses.length;
+			for (let i=0; i<length; i++) {
+				let pA = publicAddresses[i];
+				const bl = await ethereum.request({ method: "eth_getBalance", params: [pA, "latest"] });
+				// console.log("eth_getBalance: " + bl + " (" + parseInt(bl, 16) + " Ether)");
+				if (bl) {
+					console.log("bl=" + bl);
+					data.push({ 
+						pubAddres: pA,
+						bl: bl
+					});
+				}
+			}
+			setDetailData(data);
+			// console.log({ detailData: detailData });
+		}
+		fetchAccountsDetails(publicAddresses);
+	}, []);
+
+	return <TableContainer  style={{ color: "white" }}>
+		<Table>
+			<TableBody>
+				<TableRow>
+					<TableCell style={{ color: "white", borderBottom: "none" }}>Account (Public Address)</TableCell>
+					<TableCell style={{ color: "white", borderBottom: "none" }}>Balance (Eth)</TableCell>
+				</TableRow>
+				{detailData.map((value, key) => {
+					return <AccountEntry key={key} data={value}></AccountEntry>;
+				})}
+			</TableBody>
+		</Table>
+	</TableContainer>;
+}
+
+export default function Wallet({ user, setUser }) {
+	const [publicAddresses, setPublicAddresses] = useState("");
+	const history = useHistory();
+
+	const routeChange = (path) =>{
+		history.push(path);
+	};
+
+	useEffect(() => {
+		appwriteApi.getAccount()
+			.then(res => setUser(res));
+	}, []);
+
+	if (!user){
```

Comment:
What is `user`? A global variable and why? This name does not sound like a boolean so the intension is not clear. Should this be a `null` check? You can add a self-describing name like `const hasLoggedInUser = (user != null)`.